python/foglamp/plugins/north/pi_server/pi_server.py

```diff
@@ -799,28 +795,37 @@ def _generate_omf_typename_automatic(self, asset_code):
                         status_code = resp.status
                         text = await resp.text()
 
-            except Exception as e:
-                _error = Exception(plugin_common.MESSAGES_LIST["e000024"].format(e))
-                _message = plugin_common.MESSAGES_LIST["e000024"].format(e)
+            except Exception as ex:
+
+                # The conversion to a string of the exception TimeoutError produces an empty string,
+                # so it forces a proper message
```

Comment:
I think the current implementation could be fine, we test for  : 
{code}
if str(ex) == ""
{code}

that occur for TimeoutError  but it could occur for any type of exception, 
so the current code seems more general to me.

btw, if you think differently I will change.

Reply:
here the code block in try is aiohttp client sesion with timeout, so let's expect TimeoutError explicitly, other exception should be caught by except Exception .